### PR TITLE
feat: [STO-5550]: Security Steps [STO-5555]  [STO-5554]  [STO-5553]  [STO-5551] [STO-5678] [STO-5679] [STO-5680]

### PR DESCRIPTION
**Pull request by jsentongo**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-sto/components/PipelineSteps/MetasploitStep/MetasploitStepBase.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+            <SecurityScanFields
+              allowableTypes={allowableTypes}
+              formik={formik}
+              stepViewType={stepViewType}
+              scanModeSelectItems={scanModeSelectItems}
+              scanConfigSelectItems={[
+                METASPLOIT_WEAK_SSH_CONFIG,
+                METASPLOIT_OPEN_SSL_HEARTBLEED_CONFIG,
+                METASPLOIT_DYNAMIC_BY_CVE_CONFIG
```
Comment (r0610205) [inline src/modules/85-sto/components/PipelineSteps/MetasploitStep/MetasploitStepBase.tsx:124]: Remove Dynamic by CVE

File: src/modules/85-sto/components/PipelineSteps/BrakemanStep/BrakemanStep.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    spec: {
+      mode: 'orchestration',
+      config: 'default',
+      target: {
+        type: 'repository',
+        name: '',
+        variant: '',
+        workspace: ''
+      },
+      auth: {
```
Comment (r0610205) [inline src/modules/85-sto/components/PipelineSteps/BrakemanStep/BrakemanStep.tsx:63]: Remove auth

File: src/modules/85-sto/components/PipelineSteps/NiktoStep/NiktoStepBase.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+              }}
+              formik={formik}
+            />
+
+            <SecurityScanFields
+              allowableTypes={allowableTypes}
+              formik={formik}
+              stepViewType={stepViewType}
+              scanModeSelectItems={scanModeSelectItems}
+              scanConfigSelectItems={[NIKTO_DEFAULT_CONFIG, NIKTO_FULL_CONFIG, NIKTO_FULL_WEB_CONFIG]}
```
Comment (r0610205) [inline src/modules/85-sto/components/PipelineSteps/NiktoStep/NiktoStepBase.tsx:121]: We just want to leave default

File: src/modules/85-sto/components/PipelineSteps/MetasploitStep/MetasploitStepBase.tsx (hunk 1 above)
Comment (r0610205) [reply src/modules/85-sto/components/PipelineSteps/MetasploitStep/MetasploitStepBase.tsx:124]: If there is a default one let's add it as well

File: src/modules/85-sto/components/PipelineSteps/OWASPStep/OWASPStep.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    spec: {
+      mode: 'orchestration',
+      config: 'default',
+      target: {
+        type: 'repository',
+        name: '',
+        variant: '',
+        workspace: ''
+      },
+      auth: {
```
Comment (r0610205) [inline src/modules/85-sto/components/PipelineSteps/OWASPStep/OWASPStep.tsx:63]: There shouldn't be auth

Comment (jsentongo) [reply src/modules/85-sto/components/PipelineSteps/OWASPStep/OWASPStep.tsx:63]: removed

File: src/modules/85-sto/components/PipelineSteps/BrakemanStep/BrakemanStep.tsx (hunk 2 above)
Comment (jsentongo) [reply src/modules/85-sto/components/PipelineSteps/BrakemanStep/BrakemanStep.tsx:63]: removed

File: src/modules/85-sto/components/PipelineSteps/NiktoStep/NiktoStepBase.tsx (hunk 3 above)
Comment (jsentongo) [reply src/modules/85-sto/components/PipelineSteps/NiktoStep/NiktoStepBase.tsx:120]: done

**Review by shaurya-harness — APPROVED** [review]

**Review by ahokinson — APPROVED** [review]

**Review by r0610205 — APPROVED** [review]

**Pull request merged**